pages/publicLibraryFiles/PublicLibraryFileHandler.inc.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * @defgroup controllers_api_file File API controller
+ */
+
+/**
+ * @file controllers/api/file/FileApiHandler.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PublicLibraryFileHandler
+ * @ingroup pages_publicLibraryFiles
+ *
+ * @brief Class defining a handler for public library file access
+ */
+
+// Import the base handler.
+import('classes.handler.Handler');
+
+class PublicLibraryFileHandler extends Handler {
```

Comment:
One option would be to move the authenticated download out of the FileApiHandler, into the new handler -- that way they would both be together.

Reply:
@asmecher I made a change for which I am not entirely sure it covered your recommendation. Both download functions are in one Handler, but I kept a function in the api that calls the new "common" handler for download. (so that it can be used here `https://github.com/pkp/pkp-lib/blob/master/controllers/api/file/linkAction/DownloadLibraryFileLinkAction.inc.php#L34` )